src/ecosystem/corals/growth.jl

```diff
@@ -247,6 +272,194 @@ function bleaching_mortality!(Y::AbstractArray{Float64,2},
+           locations
+- `dist_t1` : Critical DHW threshold distribution for next timestep, for all species and
+              locations
+- `prop_mort` : Cache to store records of bleaching mortality
+
+# References
+1. Bairos-Novak, K.R., Hoogenboom, M.O., van Oppen, M.J.H., Connolly, S.R., 2021.
+     Coral adaptation to climate change: Meta-analysis reveals high heritability across
+     multiple traits.
+   Global Change Biology 27, 5694-5710.
+   https://doi.org/10.1111/gcb.15829
+
+2. Hughes, T.P., Kerry, J.T., Baird, A.H., Connolly, S.R., Dietzel, A., Eakin, C.M.,
+     Heron, S.F., Hoey, A.S., Hoogenboom, M.O., Liu, G., McWilliam, M.J., Pears, R.J.,
+     Pratchett, M.S., Skirving, W.J., Stella, J.S., Torda, G., 2018.
+   Global warming transforms coral reef assemblages.
+   Nature 556, 492-496.
+   https://doi.org/10.1038/s41586-018-0041-2
+
+3. Álvarez-Noriega, M., Baird, A.H., Bridge, T.C.L., Dornelas, M., Fontoura, L.,
+     Pizarro, O., Precoda, K., Torres-Pulliza, D., Woods, R.M., Zawada, K.,
+     Madin, J.S., 2018.
+   Contrasting patterns of changes in abundance following a bleaching event between
+     juvenile and adult scleractinian corals.
+   Coral Reefs 37, 527-532.
+   https://doi.org/10.1007/s00338-018-1677-y
+
+4. Baird, A., Madin, J., Álvarez-Noriega, M., Fontoura, L., Kerry, J., Kuo, C.,
+     Precoda, K., Torres-Pulliza, D., Woods, R., Zawada, K., & Hughes, T. (2018).
+   A decline in bleaching suggests that depth can provide a refuge from global
+     warming in most coral taxa.
+   Marine Ecology Progress Series, 603, 257-264.
+   https://doi.org/10.3354/meps12732
+"""
+function bleaching_mortality!(cover::Matrix{Float64}, dhw::Vector{Float64},
+    depth_coeff::Vector{Float64}, dist::Matrix{Truncated}, dist_t1::Matrix{Truncated},
+    prop_mort::SubArray{Float64})::Nothing
+    n_sp_sc, n_locs = size(cover)
+
+    # Adjust distributions for all locations, ignoring juveniles
+    # we assume the high background mortality of juveniles
+    @floop for (sp_sc, loc) in Iterators.product(3:n_sp_sc, 1:n_locs)
+        # Skip if location experiences no heat stress or there is no population
+        if dhw[loc] == 0.0 || cover[sp_sc, loc] == 0.0
+            continue
+        end
+
+        affected_pop::Float64 = cdf(dist[sp_sc, loc], dhw[loc])
+        mort_pop::Float64 = 0.0
+        if affected_pop > 0.0
+            # Calculate depth-adjusted bleaching mortality
+            mort_pop = affected_pop * depth_coeff[loc]
+
+            # Set values close to 0.0 (e.g., 1e-214) to 0.0
+            # https://github.com/JuliaLang/julia/issues/23376#issuecomment-324649815
+            if (mort_pop + one(mort_pop)) ≈ one(mort_pop)
+                mort_pop = 0.0
+            end
+        end
+
+        prop_mort[sp_sc, loc] = mort_pop
+        if mort_pop > 0.0
+            # Re-create distribution
+            d = censored(dist[sp_sc, loc]; lower=mort_pop)
```

Comment:
Maybe add a note saying this replaces values above the upper limit with the upper limit , just for future reference

Reply:
Not sure it matters here - it's a censored distribution of TruncatedNormals (there will be no values above the upper limit), and it's used temporarily to generate a new TruncatedNormal for `dist_t1`.

Reply:
Sorry, it's used with the lower limit not upper... if you think it doesn't matter, all good